connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -247,6 +248,19 @@ public boolean isSinkConnector(String connName) {
         }
     }
 
+    /*
+     * Retrieves the connector type by connector name
+     *
+     * @return the ConnectorType for the given connector name.
+     *  null if the worker does not manage a connector with the given name.
+     */
+    public ConnectorType getConnectorType(String connName) {
+        WorkerConnector workerConnector = connectors.get(connName);
+        if (workerConnector == null)
+            return null;
+        return isSinkConnector(connName) ? ConnectorType.SINK : ConnectorType.SOURCE;
+    }
+
```

Comment:
Okay, so I'm working on some metrics, and it'd be really nice to be able to get the ConnectorType off of the `WorkerConnector`, which already has a `isSinkConnector()` method. What do you think about putting the bulk of the logic there, and then having the last line of this method be `return workerConnector.getConnectorType();`?

Reply:
Such refactoring w.r.t. WorkerConnector can be done in your PR.

What do you think ?

Reply:
The reason I think it's beneficial here is that the `Worker.isSinkConnector(name)` looks up the connector by name, which you've already done 3 lines before that. And `Worker.isSinkConnector(String)` ultimately calls the `WorkerConnector.isSinkConnector`. Seems beneficial to move the tertiary check and to `WorkerConnector.getConnectorType()`.

Reply:
How about adding isSinkConnector(WorkerConnector) to Worker so that the look up is not repeated ?
isSinkConnector(String) and getConnectorType(String) call this new method.